### DISCUSSION TITLE
What is `priority` in `registerFileSystemOverlay`, if there registered multiple file system overlay, will the editor tries them one by one?

### DISCUSSION THREAD

**Discussion by alissa-tung** [CodinGame/monaco-vscode-api#405] [Q&A] [answered]
```
Register a file system overlay

By default, a memory filesystem is used to read and write file

This method allows to register another fileSystemProvider in front OR behind the default memory one.

The default one is registered as priority: 0, so:

any provider registered with a positive priority will be in front of the default one
any provider registered with a negative priority will be behind the default one
```

Hello, thank you for develop this library. I am wondering, what is `priority` in `registerFileSystemOverlay`, if there registered multiple file system overlay, will the editor tries them one by one? If so, what is the method to trigger editor to try another overlay?

**Comment by CGNonofr** [comment, marked as answer]
If an overlay throws a `FileSystemProviderError` with code `NoPermissions`, `FileNotFound` or `Unavailable`, it continues to the next overlay, otherwise the error is thrown

**Reply by alissa-tung** [reply]
Thank you! I think we can add it to code docs of this function, I am willing to help but I am not sure with the details. So I am not going to file a PR now.
